spells.yml

```diff
@@ -10574,17 +10575,15 @@ wall_of_flesh:
   components: V, S, M
   concentration: true
   description: |-
-    You create a wall of living flesh within range. The wall is 1 foot thick and is composed of ten 10-foot-by-10-foot panels. Each panel must be contiguous with at least one other panel, be vertically oriented, and be supported underneath by a solid surface or on any sides by at least two other panels. If the wall cuts through a creature's space when it appears, the creature is pushed to one side of the wall (your choose which side).
-
-    Each panel is a Large object with 10 AC, 30 hit points, and immunity to psychic damage and forced movement. A panel is destroyed if it is reduced to 0 hit points, potentially causing other panels to collapse and be destroyed.
+    You create a wall of living flesh within range. The wall is 1 foot thick, up to 10 feet tall, and up to 40 feet long in a straight line. It must be supported underneath by a solid surface at all times. If the wall cuts through a creature's space when it appears, the creature is pushed to one side of the wall (you choose which side).
 
-    At the start of each of your turns, you can cause the wall to regenerate up to 30 hit points, divided as you choose among any number of panels. Alternatively, you can choose a panel that has all its hit points and cause a new panel to grow from it. The new panel must be connected to the old panel but otherwise follows the rules of panel creation as above.
+    The wall is an object with 10 AC and immunity to psychic damage and forced movement. It starts with 2 hit points for every foot of its length, regardless of its height. The wall has no hit point maximum. Whenever it gains or loses hit points, the wall must grow or shrink so that its length in feet is equal to half its current hit points. You determine the amount by which it grows (or shrinks) in each direction, potentially pushing creatures lengthwise according to the rules in the next paragraph. The wall can't gain hit points if there isn't space for it to grow.
```

Comment:
This feels like it could bog down combat if it has to resize itself after every instance of taking damage. Perhaps it only resizes itself at either the start or end of your turn?

Reply:
Immediate resizing is necessary to be able to break through it. Would it be better if its length was rounded to the nearest 5 feet?

Reply:
Can I mark this as resolved for now, and we can fix it later if it does bog down combat?

Reply:
That's fine by me.